apps/utils/db-query-checker/src/lib.rs

```diff
@@ -0,0 +1,433 @@
+                    count_success = count_success_current,
+                    count_failure = count_failure.load(std::sync::atomic::Ordering::Relaxed),
+                    "Still running parameter sets."
+                );
+            }
+        }
+    }
+
+    // We're all done! Report overall results.
+    info!(
+        count_success = count_success.load(std::sync::atomic::Ordering::SeqCst),
+        count_failure = count_failure.load(std::sync::atomic::Ordering::SeqCst),
+        "Application complete!"
+    );
+    Ok(())
+}
+
+/// Runs all of the DB queries for the BFD Server's "search for Patients by Part D contract and year-month"
+/// endpoint, for the specfied parameters.
+///
+/// Parameters:
+/// * `db_pool`: The database connection pool to run the query on.
+/// * `csv_serializer`: The CSV [AsyncSerializer] to output results to.
+/// * `partd_contract_id`: The Part D contract ID to run the DB queries for.
+/// * `year_month`: The "YYYY-MM-dd" date to run the DB queries for.
+#[tracing::instrument(level = "debug", skip(db_pool, csv_serializer))]
+async fn check_contract_year_month(
+    db_pool: &Pool<Postgres>,
+    csv_serializer: Arc<Mutex<AsyncSerializer<File>>>,
+    partd_contract_id: String,
+    year_month: String,
+) -> Result<()> {
+    let bene_count = select_bene_count_for_part_d_contract_id_and_year_month(
+        db_pool,
+        csv_serializer.clone(),
+        &partd_contract_id,
+        &year_month,
+    )
+    .await?;
+    if bene_count == 0 {
+        return Ok(());
+    }
+
+    let bene_ids_first_page = select_bene_ids_by_part_d_contract_id_and_year_month(
+        db_pool,
+        csv_serializer.clone(),
+        &partd_contract_id,
+        &year_month,
+    )
+    .await?;
+
+    select_bene_records_by_bene_ids(db_pool, csv_serializer.clone(), &bene_ids_first_page).await?;
+
+    let mut bene_id_max = bene_ids_first_page.last().cloned();
+    while let Some(bene_id_max_value) = bene_id_max {
+        let bene_ids_next_page =
+            select_bene_ids_by_part_d_contract_id_and_year_month_and_min_bene_id(
+                db_pool,
+                csv_serializer.clone(),
+                &partd_contract_id,
+                &year_month,
+                &bene_id_max_value,
+            )
+            .await?;
+
+        select_bene_records_by_bene_ids(db_pool, csv_serializer.clone(), &bene_ids_next_page)
+            .await?;
+
+        bene_id_max = bene_ids_next_page.last().cloned();
+    }
+
+    Ok(())
+}
+
+/// Selects the count of enrolled benes for the specified Part D Contract ID and year-month.
+///
+/// Parameters:
+/// * `db_pool`: The database connection pool to run the query on.
+/// * `csv_serializer`: The CSV [AsyncSerializer] to output results to.
+/// * `partd_contract_id`: The Part D contract ID to run the DB queries for.
+/// * `year_month`: The "YYYY-MM-dd" date to run the DB queries for.
+#[tracing::instrument(level = "trace", skip(db_pool, csv_serializer))]
```

Comment:
Not done any rust, but structure and convention is very similar to node...this looks really nice!